earthly/rust/Earthfile

```diff
@@ -0,0 +1,75 @@
+# Common Rust UDCs and Builders.
+VERSION 0.7
+
+# cspell: words rustup automake autotools xutils miri nextest kani
+
+# Base Rustup build container.
+rustup:
+    # Base it on Debian bookworm-slim
+    FROM debian:bookworm-slim
+
+    WORKDIR /root
+
+    # Install necessary packages 
+    # Expand this list as needed, rather than adding more tools in later containers.
+    RUN apt-get update && \
+        apt-get install --no-install-recommends -y \
+            ca-certificates \
+            curl \
+            file \
+            build-essential \
+            autoconf \
+            automake \
+            autotools-dev \
+            libtool \
+            xutils-dev \
+            mold \
+            python3 \
+            python3-pip \
+            musl-tools && \
+        rm -rf /var/lib/apt/lists/*
+
+    # Install OpenSSL here but ONLY if we can't avoid using it.
+
+    # Install rustup.
+    RUN curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain none -y -v
```

Comment:
This isn't a deal breaker, it is more of an FYI. These `curl | bash` install methods are pretty risky inside of a build environment. Think of it from the role of an attacker: they can clearly see we use `rustup` and are using the default install method. If they manage to take ownership of the HTTP response (legitimately or illegitimately), we're effectively running an unknown payload with root permissions. 

This might seem innocent, given this is a builder that gets thrown away...but it's not far-fetched to infect a build environment and start maliciously manipulating builds. 

Again, not trying to be pedantic, but if there's an alternative install method, or at the very least a signature we can use to validate the response...that would be nice.

Reply:
I agree, but they don't have a way I can find to validate it beyond relying on https to validate the domain.  Which I have read replies to this topic suggesting that should be enough.
The official rust docker image does this, it only adds a SHA256 hash check, which they hard code.
We can't do that easily because we want to align the rust version with the `rust-toolchain.toml` of the project being built.
And I can't find a way to get an authoritative list of the hashes, and if I could we would still be back to relying on https to ensure the hash list is accurate, so its a catch-22.
The best would be if they signed it, and we knew the public key.  But they don't.

The easiest supply chain hack on rust would be to hack their docker github repo, point it to a malicious version, and re-release a malware package.  It looks like they use github CI to sign and publish, so that would still sign and publish the malware.

Reply:
> We can't do that easily because we want to align the rust version with the rust-toolchain.toml of the project being built.

Not sure I'm following. The script itself likely remains fairly static. What's preventing us from doing the same thing: hashing a known version and embedding it in the Earthfile. It's a pain to update, but like I said, I wouldn't imagine it changing _that_ much.

Reply:
Well, then why not just put `rustup.sh` in the `/earthly/rust/` directory and copy it locally.  No need to download or hash check at all.
Debian has added `rustup` package, but its unstable.  Otherwise we could install it like anything else:
https://ftp-master.debian.org/new.html

Reply:
I Just triple-checked.  It doesn't do any validation.  100% relies on https/tls1.3 to be secure to ensure it's getting the correct files.  So, it's no less secure than what we do here.  Changing it just adds maintenance work with no increase in security.  It doesn't even hash check anything.

Reply:
Yeah, not really surprising. Maybe I'll file an upstream issue if there isn't already one.